fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/DataSourceConstraint.java

```diff
@@ -54,6 +54,22 @@ public DataSourceConstraint(DataApiRequest dataApiRequest, DruidAggregationQuery
         ).collect(Collectors.toSet()));
     }
 
+    /**
+     * Copy Constructor.
+     *
+     * @param dataSourceConstraint  The data source constarint to copy from
+     */
+    protected DataSourceConstraint(DataSourceConstraint dataSourceConstraint) {
+        this.requestDimensions = dataSourceConstraint.getRequestDimensions();
+        this.filterDimensions = dataSourceConstraint.getFilterDimensions();
+        this.metricDimensions = dataSourceConstraint.getMetricDimensions();
+        this.metricNames = dataSourceConstraint.getMetricNames();
+        this.apiFilters = dataSourceConstraint.getApiFilters();
+        this.allDimensions = dataSourceConstraint.getAllDimensions();
+        this.allDimensionNames = this.getAllDimensionNames();
+        this.allColumnNames = this.getAllColumnNames();
+    }
```

Comment:
In general, `protected` and `private` constructors like this are relying on the fact that the external interfaces are ensuring the collections like this are immutable, so there's usually no need to defensively copy collections for those sorts of copy-constructors.

tldr: In this case, it's OK to use the item directly, not need to defensively copy